src/jobs/activities/wallet-activity-tracking.ts

```diff
@@ -32,6 +34,39 @@ export const walletTransactionLogsQueue = new Queue(
   queueOptions
 );
 
+const getCacheKey = (accountId: string, txHash: string) => `${accountId}:${txHash}`;
+
+const invokeWebhookEndpoints = async (
+  transferEvent: TransferEventData,
+  payload: WalletActivityEvent
+) => {
+  const txHash = transferEvent.transactionHash;
+
+  const accountTransactionCache = new Map<string, WebhookRequest>();
+  const webhookRequestsForSender = await getWebhookRequestsForAddress(transferEvent.fromAddress);
+  const webhookRequestsForRecipient = await getWebhookRequestsForAddress(transferEvent.toAddress);
+
+  for (const webhookRequest of webhookRequestsForSender) {
+    const { accountId } = webhookRequest;
+    const cacheKey = getCacheKey(accountId, txHash);
+    accountTransactionCache.set(cacheKey, { ...webhookRequest, data: payload });
+  }
+
+  for (const webhookRequest of webhookRequestsForRecipient) {
+    const { accountId } = webhookRequest;
+    const cacheKey = getCacheKey(accountId, txHash);
+    accountTransactionCache.set(cacheKey, { ...webhookRequest, data: payload });
+  }
+
+  const webhookRequests = accountTransactionCache.values();
```

Comment:
We get unique records for txHash and accounts(env). Each env will get notified of a user_transaction only once even if both **from_address** and **to_address** are being tracked in an environment.